src/UI/DappModal/components/DappModal.tsx

```diff
@@ -0,0 +1,78 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import icons from '../../../optionalPackages/fortawesome-free-solid-svg-icons';
+import ReactFontawesome from '../../../optionalPackages/react-fontawesome';
+import styles from '../styles/dapp-modal.scss';
+import { DappModalConfig } from '../types';
+
+type AwesomeModalProps = {
+  visible: boolean;
+  onHide: () => void;
+  parentElement?: Element;
+  children?: React.ReactNode;
+} & DappModalConfig;
+
+const DappModal: React.FC<AwesomeModalProps> = ({
+  visible,
+  onHide,
+  showHeader,
+  showFooter,
+  headerText,
+  footerText,
+  modalDialogClassName,
+  modalContentClassName,
+  modalHeaderClassName,
+  modalCloseButtonClassName,
+  modalBodyClassName,
+  modalFooterClassName,
+  parentElement,
+  children
+}) => {
+  return visible
+    ? ReactDOM.createPortal(
+        <div
+          id='dapp-modal'
+          role='dialog'
+          aria-modal='true'
+          className={`${styles.dappModal} ${modalDialogClassName}`}
+          onClick={onHide}
+        >
+          <div
+            className={`${styles.dappModalContent} ${modalContentClassName}`}
+          >
+            {showHeader && (
```

Comment:
Maybe we should pull this out into a named constant: const headerJSX = showHeader ?